### DISCUSSION TITLE
Initialize React.createRef with a value in Zustand store

### DISCUSSION THREAD

**Discussion by martinaedma** [pmndrs/zustand#1441] [Q&A]
store has a property

```
scrollRef: RefObject<ScrollRef>;
```

ScrollRef is defined like so

```
interface ScrollRef {
  value: number
}
```

Now how do I initialize it in create method inside store ? with a default value of 0

when I write
```
 scrollRef: createRef()
```

this gives no errors but scrollRef is undefined when used in components.
I tried some syntactic options but failed to initialize with a default value of 0.
How do I do that ?

**Comment by dai-shi** [comment]
I think it's the same as how you do with a global variable.
```js
const ref = createRef()
```
